### DISCUSSION TITLE
Announce when joining a channel

### DISCUSSION THREAD

**Discussion by AnotherGlitch** [codetheweb/muse#924] [Ideas]
In our server, so many people greet our music bot. It would be nice to be able to point to a local file (or a link I suppose), and add that as an intro when Muse joins a channel.

I've looked through the code but, I am lacking in TypeScript knowledge. My first guess would be to import the file and add it to the 0 queue position within `async connect` portion of the player.ts script. Though my gut says a `/config set-intro` boolean slash command, defaulting to false, would be necessary for anyone not wanting this functionality. 

I generally run a docker container, so I imagine a local folder could be mapped in the docker compose file pointing to an .mp3 or other audio source. Though, not sure what trouble this might cause for anyone not using docker. Maybe a file location in one of the config files would work better?

Thoughts/ideas/advice?

**Comment by codetheweb** [comment]
I don't think this will ever be in Muse, but you definitely fork and add it there. :) Adding a new track to the front of the queue is a great plan.
